Fixes #3070: tests for Funds/updater.ts

Faisalmajeed123 opened the pull request:
<!--
This section can be deleted after reading.

We employ the following branching strategy to simplify the development process and to ensure that only stable code is pushed to the `master` branch:

- `develop`: For unstable code: New features and bug fixes.
- `master`: Where the stable production ready code lies. Only security related bugs.

NOTE!!!

ONLY SUBMIT PRS AGAINST OUR `DEVELOP` BRANCH. THE DEFAULT IS `MAIN`, SO YOU WILL HAVE TO MODIFY THIS BEFORE SUBMITTING YOUR PR FOR REVIEW. PRS MADE AGAINST `MAIN` WILL BE CLOSED.

-->

<!--
Thanks for submitting a pull request! Please provide enough information so that others can review your pull request.
-->

**What kind of change does this PR introduce?**

Test Coverage

**Issue Number:**

Fixes #3070 

**If relevant, did you update the documentation?**

N/A 

**Summary**

Added tests for src/graphql/types/Fund/updater.ts

**Does this PR introduce a breaking change?**

No

## Checklist

### CodeRabbit AI Review
- [ ] I have reviewed and addressed all critical issues flagged by CodeRabbit AI
- [ ] I have implemented or provided justification for each non-critical suggestion
- [ ] I have documented my reasoning in the PR comments where CodeRabbit AI suggestions were not implemented

### Test Coverage
- [ ] I have written tests for all new changes/features
- [ ] I have verified that test coverage meets or exceeds 95%
- [ ] I have run the test suite locally and all tests pass


**Other information**

<!--Add extra information about this PR here-->

**Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa-api/blob/master/CONTRIBUTING.md)?**

Yes


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **Refactor**
	- Enhanced the authentication and user resolution process for fund updates, ensuring accurate user verification and improved error handling.
- **Tests**
	- Introduced a comprehensive suite of unit tests covering scenarios such as unauthorized access, null updater values, and unexpected errors to ensure robust and reliable system behavior.
	- Added new interfaces to support testing scenarios, enhancing type safety and clarity.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

Faisalmajeed123 commented:
@coderabbitai full review

Faisalmajeed123 commented:
should I implement this or not kindly confirm:
```
- if (
-   currentUser.role !== "administrator" &&
-   (currentUserOrganizationMembership === undefined ||
-    currentUserOrganizationMembership.role !== "administrator")
- ) {
+ const isAdminInAnyOrg =
+   currentUser.organizationMembershipsWhereMember?.some(
+     (membership) => membership.role === "administrator"
+   );
+ if (currentUser.role !== "administrator" && !isAdminInAnyOrg) {
    throw new TalawaGraphQLError({
      extensions: {
        code: "unauthorized_action",
      },
    });
  }
  ```

palisadoes commented:
> should I implement this or not kindly confirm:
> 
> ```
> - if (
> -   currentUser.role !== "administrator" &&
> -   (currentUserOrganizationMembership === undefined ||
> -    currentUserOrganizationMembership.role !== "administrator")
> - ) {
> + const isAdminInAnyOrg =
> +   currentUser.organizationMembershipsWhereMember?.some(
> +     (membership) => membership.role === "administrator"
> +   );
> + if (currentUser.role !== "administrator" && !isAdminInAnyOrg) {
>     throw new TalawaGraphQLError({
>       extensions: {
>         code: "unauthorized_action",
>       },
>     });
>   }
> ```

Keep the current logic

palisadoes commented:
1. Please make coderabbit.ai approves your work.
2. Make sure all tests pass and are valid.
3. Ensure the test code coverage for your patch reaches as close to 100% as possible.

Faisalmajeed123 commented:
@coderabbitai full review

Faisalmajeed123 commented:
@coderabbitai full review

Faisalmajeed123 commented:
https://github.com/coderabbitai full review

Faisalmajeed123 commented:
@coderabbitai full review

Faisalmajeed123 commented:
@palisadoes please review

palisadoes reviewed (changes requested):
The aim is to write tests on an unmodified source file, not remove functionality. The only exception is if you can prove that the source file needs modification

1. You have removed functionality. Please restore
    ![image](https://github.com/user-attachments/assets/c74e3273-d108-4aac-9a94-13198215dc9b)
1. You have removed functionality. Please restore
    ![image](https://github.com/user-attachments/assets/dfa06ef2-1503-44e3-be55-0191847b8170)
1. You have removed functionality. Please restore
    ![image](https://github.com/user-attachments/assets/bfc5ac18-ee65-4ed4-8a7b-92ee553f568b)

Faisalmajeed123 commented:
I understand your concern about removing functionality. However, my changes do not alter the logic or behaviour of the code.

1- I extracted authentication logic into a separate function (authenticateUser) to improve reusability.
2- I moved the resolver logic to resolveUpdater, keeping all the checks intact.
3- No functionality has been removed; it has just been refactored for clarity and maintainability.

If you still believe something is missing, I’d appreciate your guidance on what needs to be restored. Please let me know if you’d like me to make any adjustments.

first:
<img width="370" alt="chrome_EhHb0ZxVsI" src="https://github.com/user-attachments/assets/0411eef5-6d55-4f98-b430-28f0f9302a5f" />
<img width="367" alt="chrome_e2PQnuRaNP" src="https://github.com/user-attachments/assets/2379c40f-2505-4399-af96-5c7ce7217d3f" />

Second:
regarding the "organizationMembershipsWhereMember" filter. Please confirm whether to restore it or leave it as it is.
<img width="373" alt="chrome_msVmO0ZUXh" src="https://github.com/user-attachments/assets/bbdd6ef5-c8c9-4e06-8642-c43a521215e3" />
<img width="377" alt="chrome_z976qqSZzJ" src="https://github.com/user-attachments/assets/56f83608-7b72-495a-ade0-d825835ff300" />

Third:
<img width="351" alt="chrome_JgYTAdGfSB" src="https://github.com/user-attachments/assets/1a0221fd-5e2d-4b2d-8f89-bfc01b2ab127" />
<img width="368" alt="chrome_lX9o56EBon" src="https://github.com/user-attachments/assets/fd31d570-9749-43e7-ae48-0996a0e3b38f" />
